lib/Dialects/LinalgTransform/Transforms/PDL.cpp

```diff
@@ -0,0 +1,84 @@
+//===-- PDL.cpp - Interoperability with PDL -------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PDL.h"
+
+#include "Transforms/Functional.h"
+#include "mlir/Dialect/PDL/IR/PDLOps.h"
+#include "mlir/Rewrite/FrozenRewritePatternSet.h"
+#include "mlir/Rewrite/PatternApplicator.h"
+
+namespace mlir {
+namespace linalg {
+
+/// Return ops that match any of the patterns.
+static SmallVector<LinalgOp> getMatchingOps(
+    Operation *parent, const FrozenRewritePatternSet &patterns) {
+  PatternApplicator applicator(patterns);
+  applicator.applyDefaultCostModel();
+
+  // TODO: The C++ functional API needs better interoperability with PDL.
+  return functional::applyForEachIn(
+      parent,
+      [&](Operation *op, PatternRewriter &rewriter) -> FailureOr<LinalgOp> {
+        if (succeeded(applicator.matchAndRewrite(op, rewriter)))
+          if (auto linalgOp = dyn_cast<LinalgOp>(op)) return linalgOp;
```

Comment:
I'm surprised clang-format accepted this, I would expect the return statement to go on the next line.